.github/workflows/e2e.yml

```diff
@@ -0,0 +1,71 @@
+name: "e2e"
+
+on:
+  pull_request:
+    paths-ignore:
+      - "**.md"
+      - ".github/**"
+      - "!.github/workflows/e2e.yml"
+      - "!.github/workflows/snapshot.yml"
+    types:
+      - "labeled"
+      - "synchronize"
+      - "opened"
+      - "reopened"
+  push:
+    branches:
+      - "main"
+    paths-ignore:
+      - "**.md"
+      - ".github/**"
+      - "!.github/workflows/e2e.yml"
+      - "!.github/workflows/snapshot.yml"
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  snapshot:
+    if: >
+      github.event_name == 'push' ||
+      contains(github.event.pull_request.labels.*.name, 'e2e')
+    name: "aqua-ipfs"
+    uses: ./.github/workflows/snapshot.yml
+    with:
+      ref: ${{ github.ref }}
+
+  nox:
+    needs:
+      - snapshot
+    uses: fluencelabs/nox/.github/workflows/build.yml@master
+    with:
+      cargo-dependencies: |
+        [
+          {
+            "package": "aqua-ipfs-distro",
+            "version": "=${{ needs.snapshot.outputs.cargo-version }}",
+            "manifest": "crates/system-services/Cargo.toml",
+            "registry": "fluence"
+          }
+        ]
+
+  nox-snapshot:
+    name: "nox"
+    needs:
+      - nox
```

Comment:
nox needs nox? is that a recursion?

if not, naming is bad

Reply:
@nahsi should we call it `nox-snapshot`?

Reply:
The job id is `nox-snapshot` and it needs job id `nox`. The name is set `nox` so it looks nice in github UI. Github will place all CI jobs that happen in nox repo in one tab called `nox` this way.

We could rename `nox` to `nox-binary` or something but for consistency should do it in all repos :'(